rice-framework/krad-service-impl/src/main/java/org/kuali/rice/krad/service/impl/DocumentServiceImpl.java

```diff
@@ -211,15 +211,22 @@ private DocumentEvent generateKualiDocumentEvent(Document document,
     @Override
     public Document routeDocument(Document document, String annotation,
             List<AdHocRouteRecipient> adHocRecipients) throws ValidationException, WorkflowException {
+        return routeDocument(document, annotation, null, adHocRecipients);
+    }
+
+    /**
+     * @see org.kuali.rice.krad.service.DocumentService#routeDocument(org.kuali.rice.krad.document.Document,
+     *      java.lang.String, java.util.List)
+     */
+    @Override
+    public Document routeDocument(Document document, String annotation, String nodeName,
+                                  List<AdHocRouteRecipient> adHocRecipients) throws ValidationException, WorkflowException {
         checkForNulls(document);
-        //if (!getDocumentActionFlags(document).getCanRoute()) {
-        //    throw buildAuthorizationException("route", document);
-        //}
         document.prepareForSave();
         Document savedDocument = validateAndPersistDocument(document, new RouteDocumentEvent(document));
         prepareWorkflowDocument(savedDocument);
         getWorkflowDocumentService()
-                .route(savedDocument.getDocumentHeader().getWorkflowDocument(), annotation, adHocRecipients);
+                .route(savedDocument.getDocumentHeader().getWorkflowDocument(), annotation, savedDocument.getAdHocRouteNodeName(), adHocRecipients);

```

Comment:
...continuing from above comment...

Then here you should always just use the passed in node name to pass to the associated method. You aren't actually even using the nodeName parameter in any of these methods. I would also rename the parameter to `adHocRouteNodeName` to make it more clear what it means in this context.